src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/LoggerMessageDefineAnalyzer.cs

```diff
@@ -251,6 +268,45 @@ private void AnalyzeFormatArgument(OperationAnalysisContext context, IOperation
             }
         }
 
+        /// <summary>
+        /// Does the text have valid braces? (no unclosed braces, or braces without an opening)
+        /// </summary>
+        /// <param name="text">The text to check.</param>
+        /// <returns>When true brackets are valid, false otherwise.</returns>
+        private static bool IsValidBraces(string text)
+        {
+            var stack = new Stack<char>();
```

Comment:
This is not going to work well as the `{` and `}` can be escaped. Please look at https://source.dot.net/#Microsoft.Extensions.Logging.Abstractions/LogValuesFormatter.cs,42 how these are parsed. 

CC @stephentoub

Reply:
heh... part of the problem is that `LogValuesFormatter` is what is *throwing* when passing in a bad format string from the original issue 😅 

could probably use some of the code in there as a way to bootstrap a working solution on the analyzer side - but the real problem is that now such badly formatted message templates throw with the stack trace:

```cs
   at System.ThrowHelper.ThrowFormatInvalidString(Int32 offset, ExceptionResource resource)
   at System.Text.CompositeFormat.Parse(String format)
   at Microsoft.Extensions.Logging.LogValuesFormatter..ctor(String format)
   at Microsoft.Extensions.Logging.FormattedLogValues.<>c.<.ctor>b__9_0(String f)
   at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey key, Func`2 valueFactory)
   at Microsoft.Extensions.Logging.FormattedLogValues..ctor(String format, Object[] values)
   at Microsoft.Extensions.Logging.LoggerExtensions.Log(ILogger logger, LogLevel logLevel, EventId eventId, Exception exception, String message, Object[] args)
   at Microsoft.Extensions.Logging.LoggerExtensions.Log(ILogger logger, LogLevel logLevel, String message, Object[] args)
   at Microsoft.Extensions.Logging.LoggerExtensions.LogInformation(ILogger logger, String message, Object[] args)
   at Program.<Main>$(String[] args) in C:\Users\kritn\source\repos\ConsoleApp1\ConsoleApp1\Program.cs:line 13
```

from a line like:

```cs
logger.LogInformation(@"Test: {i}{", i);
```

![image](https://github.com/dotnet/roslyn-analyzers/assets/8814983/192148dd-3eca-4ff6-8482-8159f70bf585)

Reply:
> but the real problem is that now such badly formatted message templates throw at runtime with the stack trace:

Can you clarify why this is a problem?

Reply:
a problem in the general sense - as in the reasoning for submitting the issue. Additionally having been pointed to code from your comment https://github.com/dotnet/roslyn-analyzers/pull/7286#discussion_r1561388188 that actually *is* the issue that needs fixing (or flagging through the analyzer) to avoid the run time exceptions

Reply:
To be honest, I don't perceive it as a problem. The behavior, throwing an exception at runtime, is correct. Implementing an analyzer to detect the issue at compile time serves as a helpful means of catching it early, much like we do with other issues we create analyzers for.

Reply:
hmm... i was under the impression this was new behavior - we had some (seemingly) hot path code start erroring out in prod after the .net8 update, where it wasn't in 7. unless this particular function that had a log statement just hadn't been hit in over a year.

If it's existing this way for a while though then yeah i guess it's not that big of a deal, just don't see how it couldn't be new behavior /shrug

Reply:
It was a bug which is fixed. The mal-formed message shouldn't be allowed and should throw as we do now in .NET 8.0. It was just a luck that this problem existed before and not noticed.

Reply:
believe this is taken care of now through https://github.com/dotnet/roslyn-analyzers/pull/7286/commits/7333c4d2d7dc37d7b38c6f13ead7935da98d5a43